nemo/collections/asr/parts/submodules/fast_rnnt_greedy_decoding.py

```diff
@@ -0,0 +1,440 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from nemo.core.utils.cuda_python_utils import check_cuda_python_cuda_graphs_conditional_nodes_supported
+
+check_cuda_python_cuda_graphs_conditional_nodes_supported()
+
+import contextlib
+import ctypes
+import time
+from dataclasses import dataclass, field
+from itertools import product
+from typing import List, Optional, Tuple, Union
+
+import numpy as np
+import torch
+from cuda import cuda, cudart, nvrtc
+from omegaconf import DictConfig, OmegaConf
+
+from nemo.collections.asr.modules import rnnt_abstract
+from nemo.collections.asr.parts.utils import rnnt_utils
+from nemo.collections.asr.parts.utils.asr_confidence_utils import ConfidenceMethodConfig, ConfidenceMethodMixin
+from nemo.collections.common.parts.rnn import label_collate
+from nemo.core.classes import Typing, typecheck
+from nemo.core.neural_types import AcousticEncodedRepresentation, ElementType, HypothesisType, LengthsType, NeuralType
+from nemo.utils import logging
+
+
+def ASSERT_DRV(err):
+    if isinstance(err, cuda.CUresult):
```

Comment:
Inside any function that uses cuda python, check if HAVE_CUDA_PYTHON, and if not call check_cuda_python_cuda_graphs_conditional_nodes_supported() to give users a meaningful error.

Reply:
It's not that simple, though. It's not just about whether we can `import cuda`. We also need to check that the appropriate version of cuda-python itself and the cuda driver are installed. https://github.com/NVIDIA/NeMo/pull/8191/files#diff-acab1a9f3d702862ddbe5720bfa6c7fd0a57f7c3dc0b59eb9878ed5cd1e3513aR28-R45

Maybe what you want in this case is something like https://github.com/NVIDIA/NeMo/blob/main/nemo/core/utils/k2_guard.py. We will simply expect those developing with cuda-python to do `from cuda_python_guard.cuda import ...` instead `from cuda import ...`